esp-hal/src/uart.rs

```diff
@@ -317,35 +319,29 @@ impl<TX: OutputPin, RX: InputPin> UartPins for TxRxPins<'_, TX, RX> {
 }
 
 /// UART driver
-pub struct Uart<'d, T> {
+pub struct Uart<'d, T, M> {
```

Comment:
Should here be default value? And same in `UartTx`, `UartRx`.
```suggestion
pub struct Uart<'d, T, M = Blocking> {
```